src/dbnode/storage/database.go

```diff
@@ -928,15 +934,110 @@ func (d *db) ReadEncoded(
 			xopentracing.Time("end", end),
 		)
 	}
+
 	defer sp.Finish()
+	return n.ReadEncoded(ctx, id, start, end)
+}
 
+func (d *db) WideQuery(
+	ctx context.Context,
+	namespace ident.ID,
+	query index.Query,
+	queryStart time.Time,
+	shards []uint32,
+	iterOpts index.IterationOptions,
+) ([]ident.IndexChecksum, error) { // FIXME: change when exact type known.
 	n, err := d.namespaceFor(namespace)
 	if err != nil {
 		d.metrics.unknownNamespaceRead.Inc(1)
 		return nil, err
 	}
 
-	return n.ReadEncoded(ctx, id, start, end)
+	var (
+		doneCh = make(chan struct{})
+
+		batchSize = d.opts.WideBatchSize()
+		blockSize = n.Options().IndexOptions().BlockSize()
+
+		collector          = make(chan *ident.IDBatch)
+		collectedChecksums = make([]ident.IndexChecksum, 0, 10)
+		collectorErr       error
+	)
+
+	opts, err := index.NewWideQueryOptions(queryStart, batchSize, blockSize, collector, shards, iterOpts)
```

Comment:
Can we separate out the collector from the wide query options as discussed? It seems important enough to keep as it's own thing instead of inside the options (since it's basically an in-out variable).

Reply:
Sounds good :+1: